tests/testthat/helper-dettl.R

```diff
@@ -32,3 +32,8 @@ TestDataImport <- R6::R6Class(
     }
   )
 )
+
+mock_private <- function(obj, to_mock, value) {
+  unlockBinding(to_mock, obj$.__enclos_env__$private)
```

Comment:
What is this for? Generally `unlockBinding` is a function to avoid as the R people consider it "unsafe"

Reply:
Removed this now. We can add `transformed_data` for the test without call to `unlockBinding` and after the refactor the test which needed this for `pre` and `post` load can be removed as it essentially duplicated a test we had in `test_pre_post_load`

Reply:
Thinking about move to separating file processing from object initialisation I think maybe hard to separate out if we want to keep the `reload` call which reloads from file. The object will still have to take the file path and be able to read cfg from it.